clusters/resource_cluster.go

```diff
@@ -363,38 +439,54 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 		// don't add externally added libraries, if config has no `library {}` blocks
 		return nil
 	}
-	var libraryList libraries.ClusterLibraryList
+	var libraryList libraries.LibraryList
+	libraryList.ClusterId = clusterId
 	common.DataToStructPointer(d, clusterSchema, &libraryList)
-	librariesAPI := libraries.NewLibrariesAPI(ctx, c)
-	libsClusterStatus, err := librariesAPI.ClusterStatus(clusterID)
+	libsClusterStatus, err := w.Libraries.ClusterStatusByClusterId(ctx, clusterId)
+	if err != nil {
+		return err
+	}
+	libsToInstall, libsToUninstall := libraries.GetLibrariesToInstallAndUninstall(libraryList, libsClusterStatus)
+	clusterInfo, err = clusters.GetByClusterId(ctx, clusterId)
+	err = wrapMissingClusterError(err, d.Id())
 	if err != nil {
 		return err
 	}
-	libraryList.ClusterID = clusterID
-	libsToInstall, libsToUninstall := libraryList.Diff(libsClusterStatus)
 	if len(libsToUninstall.Libraries) > 0 || len(libsToInstall.Libraries) > 0 {
 		if !clusterInfo.IsRunningOrResizing() {
-			if _, err = clusters.StartAndGetInfo(clusterID); err != nil {
+			if _, err = clusters.StartByClusterIdAndWait(ctx, clusterId); err != nil {
 				return err
 			}
 		}
 		// clusters.StartAndGetInfo() always returns a running cluster
 		// or errors out, so we just know the cluster is active.
-		err = librariesAPI.UpdateLibraries(clusterID, libsToInstall, libsToUninstall,
-			d.Timeout(schema.TimeoutUpdate))
+		// TODO: Wait needed here?
+		err = w.Libraries.UpdateAndWait(ctx, compute.Update{
+			ClusterId: clusterId,
+			Install:   libsToInstall.Libraries,
+			Uninstall: libsToUninstall.Libraries,
+		})
 		if err != nil {
 			return err
 		}
 		if clusterInfo.State == ClusterStateTerminated {
-			log.Printf("[INFO] %s was in TERMINATED state, so terminating it again", clusterID)
-			if err = clusters.Terminate(clusterID); err != nil {
+			log.Printf("[INFO] %s was in TERMINATED state, so terminating it again", clusterId)
+			if err = clusters.DeleteByClusterId(ctx, clusterId); err != nil {
```

Comment:
Are delete & terminate the same?

Reply:
Yes, https://docs.databricks.com/api/workspace/clusters/delete
<img width="876" alt="image" src="https://github.com/databricks/terraform-provider-databricks/assets/88379306/ae9d1273-bbb9-4279-9579-e5b492bae542">